crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -674,6 +681,15 @@ impl<'a> Resolver<'a> {
         }
     }
 
+    /// True if the 'distinct' keyword is allowed on parameters in this function
+    fn distinct_allowed(&self, func: &NoirFunction) -> bool {
+        if self.in_contract() {
+            !func.def.is_unconstrained && !func.def.is_open
```

Comment:
Why isn't distinct allowed on unconstrained or open functions?

Reply:
Good question. It probably should be. If I understand correctly open and unconstrained functions can form a program in same manner as "secret" contract functions or `main` functions.

Reply:
distinct is used to denote the fact that the witness indices should be unique. For unconstrained functions, there are no witness indices being produced, so we would not want it there.

For open functions, they are also similar to unconstrained functions since they get compiled down to Brillig.

Reply:
But that's still brillig wrapped in a single acir opcode no? The abi around that single opcode still impacts the developer.

Or should we change compilation such that brillig isn't wrapped in an acir opcode for these functions?  If so, this sounds like an issue to address separately.

Reply:
Actually, I guess it doesn't impact the developer because they won't care about having uniform proof public arguments.